source/slang/slang-ir-lower-buffer-element-type.cpp

```diff
@@ -914,33 +1143,552 @@ struct LoweredElementTypeContext
         TypeLoweringConfig config;
     };
 
-    IRInst* getBufferAddr(IRBuilder& builder, IRInst* loadStoreInst)
+    IRInst* getBufferAddr(IRBuilder& builder, IRInst* loadStoreInst, IRInst* baseAddr)
     {
         switch (loadStoreInst->getOp())
         {
         case kIROp_Load:
         case kIROp_Store:
-            return loadStoreInst->getOperand(0);
+            return baseAddr;
         case kIROp_StructuredBufferLoad:
         case kIROp_StructuredBufferLoadStatus:
         case kIROp_RWStructuredBufferLoad:
         case kIROp_RWStructuredBufferLoadStatus:
         case kIROp_RWStructuredBufferStore:
             return builder.emitRWStructuredBufferGetElementPtr(
-                loadStoreInst->getOperand(0),
+                baseAddr,
                 loadStoreInst->getOperand(1));
         default:
             return nullptr;
         }
     }
 
+    bool maybeTranslateTailingPointerGetElementAddress(
```

Comment:
Should this say `Trailing` instead of `Tailing`? The comment at the start of the function body implies that this is a typo.